core/base/src/main/java/de/mm20/launcher2/search/Location.kt

```diff
@@ -0,0 +1,216 @@
+package de.mm20.launcher2.search
+
+import android.content.Context
+import androidx.core.content.ContextCompat
+import de.mm20.launcher2.base.R
+import de.mm20.launcher2.icons.ColorLayer
+import de.mm20.launcher2.icons.StaticLauncherIcon
+import de.mm20.launcher2.icons.TintedIconLayer
+import kotlinx.collections.immutable.ImmutableList
+import java.time.DayOfWeek
+import java.time.Duration
+import java.time.LocalDate
+import java.time.LocalTime
+
+interface Location : SavableSearchable {
+
+    val latitude: Double
+    val longitude: Double
+    val fixMeUrl: String?
+
+    suspend fun getCategory(): LocationCategory?
```

Comment:
Hm. I get the motivation that this would reduce some overhead when just getting the icon, but at the same time, OpenStreetMaps can be edited (and therefore: corrected) by anyone. So actually, I am not a fan of storing anything else than the location ID in the database, since everything may change. I guess we can just always store the category, but then update it as other methods like `getStreet()` are called?